docarray/base_document/document.py

```diff
@@ -97,3 +99,24 @@ def __setattr__(self, field, value) -> None:
             for key, val in self.__dict__.items():
                 dict_ref[key] = val
             object.__setattr__(self, '__dict__', dict_ref)
+
+    def __eq__(self, cls) -> bool:
+        for key1,key2 in zip(self.dict(),cls.dict()):
```

Comment:
I think you need to check that self and other have the same number of field

Reply:
I will be using len() to determine whether it has the same length or not. About the key I think I will be using set() to determine if it has the same key or not.